Спринт 21

IvanPodlesnykh opened the pull request:
1. Реализована работа с БД в слое data (описаны Entity, Dao)
2. Реализованы репозиторий и интерактор для работы с БД
3. Реализовано добавление трека в БД избранных треков нажатием на кнопку на экране проигрывателя
4. Реализовано отображение избранных треков на соответствующем экране

IvanPodlesnykh committed:
Спринт 21
1. Реализована работа с БД в слое data (описаны Entity, Dao)
2. Реализованы репозиторий и интерактор для работы с БД
3. Реализовано добавление трека в БД избранных треков нажатием на кнопку на экране проигрывателя
4. Реализовано отображение избранных треков на соответствующем экране

DmitrijEgorow commented on app/src/main/java/com/ivanpodlesnykh/playlistmaker/data/db/dao/FavoriteTracksDao.kt:
```diff
@@ -0,0 +1,23 @@
+    suspend fun deleteTrack(trackId: String)
+
+    @Query("SELECT * FROM favorite_tracks_table")
+    suspend fun getAllTracks(): List<TrackEntity>
```
Room позволяет возвращать сразу Flow

DmitrijEgorow commented on app/src/main/java/com/ivanpodlesnykh/playlistmaker/data/search/SearchRepositoryImpl.kt:
```diff
@@ -17,7 +19,9 @@ import kotlinx.coroutines.withContext
 class SearchRepositoryImpl(
     private val application: Application,
```
А точно нужен именно application, какой-то более широкий интерфейс не подойдёт?

IvanPodlesnykh committed:
Спринт 21
Фиксы:
Теперь реализации репозиториев принимают в конструкторе Context, а не Application
